# Exceptions as APIs

################################################################################
# run simple square root program
python3 module/roots1.py

# try to calculate square root of (-1)
# print(sqrt(-1)) # throws error
# ZeroDivisionError: float division by zero
python3 module/roots2.py

# handle exception using try
python3 module/roots3.py

# use try while calling all functions in main()
python3 module/roots4.py

# handle exception inside sqrt() function where exception occurs
# raise ValueError
python3 module/roots5.py

# detect exception condition early on & raise exception before function execution
python3 module/roots6.py

# add exceptioin handing while calling function in main()
python3 module/roots7.py

################################################################################
